tools/gfx-unit-test/shared-textures-tests.cpp

```diff
@@ -0,0 +1,223 @@
+        auto intBufferView = createOutBufferView(dstDevice, intResults);
+
+        ITextureResource::Size size = {};
+        size.width = 2;
+        size.height = 2;
+        size.depth = 1;
+
+        ITextureResource::Size bcSize = {};
+        bcSize.width = 4;
+        bcSize.height = 4;
+        bcSize.depth = 1;
+
+        {
+            float texData[] = { 1.0f, 0.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f,
+                                0.0f, 0.0f, 1.0f, 1.0f, 0.5f, 0.5f, 0.5f, 1.0f };
+            ITextureResource::SubresourceData subData = { (void*)texData, 32, 0 };
+
+            // Create a shareable texture using srcDevice, get its handle, then create a texture using the handle using
+            // dstDevice. Read back the texture and check that its contents are correct.
+            auto srcTexture = createTexture(srcDevice, size, gfx::Format::R32G32B32A32_FLOAT, &subData);
+
+            InteropHandle sharedHandle;
+            GFX_CHECK_CALL_ABORT(srcTexture->getSharedHandle(&sharedHandle));
+            ComPtr<ITextureResource> dstTexture;
+            size_t sizeInBytes = 0;
+            size_t alignment = 0;
+            GFX_CHECK_CALL_ABORT(srcDevice->getTextureAllocationInfo(*(srcTexture->getDesc()), &sizeInBytes, &alignment));
+            GFX_CHECK_CALL_ABORT(dstDevice->createTextureFromSharedHandle(sharedHandle, *(srcTexture->getDesc()), sizeInBytes, dstTexture.writeRef()));
+            // Reading back the buffer from srcDevice to make sure it's been filled in before reading anything back from dstDevice
+            // TODO: Implement actual synchronization (and not this hacky solution)
+            compareComputeResult(
+                dstDevice,
+                dstTexture,
+                ResourceState::ShaderResource,
+                texData,
+                sizeof(texData));
+
+            auto texView = createTexView(dstDevice, dstTexture);
+            setUpAndRunShader(dstDevice, dstTexture, texView, floatBufferView, "copyTexFloat4");
+            compareComputeResult(
+                dstDevice,
+                floatResults,
+                Slang::makeArray<float>(1.0f, 0.0f, 0.0f, 1.0f, 0.0f, 1.0f, 0.0f, 1.0f,
+                                        0.0f, 0.0f, 1.0f, 1.0f, 0.5f, 0.5f, 0.5f, 1.0f));
+        }
+    }
+
+    void sharedTextureTestAPI(UnitTestContext* context, Slang::RenderApiFlag::Enum srcApi, Slang::RenderApiFlag::Enum dstApi)
+    {
+        auto srcDevice = createTestingDevice(context, srcApi);
+        auto dstDevice = createTestingDevice(context, dstApi);
+        if (!srcDevice || !dstDevice)
+        {
+            SLANG_IGNORE_TEST;
+        }
+
+        sharedTextureTestImpl(srcDevice, dstDevice, context);
+    }
+#if SLANG_WIN64
+    SLANG_UNIT_TEST(sharedTextureD3D12ToCUDA)
+    {
+        sharedTextureTestAPI(unitTestContext, Slang::RenderApiFlag::D3D12, Slang::RenderApiFlag::CUDA);
+    }
+#endif
+
+#if SLANG_WINDOWS_FAMILY // TODO: Remove when Linux support is added
```

Comment:
This needs to be SLANG_WIN64.

Reply:
Woops, accidentally flipped the WINDOWS_FAMILY and WIN64 checks.